llama_index/vector_stores/postgres.py

```diff
@@ -122,23 +123,39 @@ def add(self, embedding_results: List[NodeWithEmbedding]) -> List[str]:
         return ids
 
     def _query_with_score(
-        self, embedding: Optional[List[float]], limit: int = 10
+        self,
+        embedding: Optional[List[float]],
+        limit: int = 10,
+        metadata_filters: Optional[MetadataFilters] = None,
     ) -> List[Any]:
+        import sqlalchemy
+        from sqlalchemy import and_
+
         with self._session() as session:
             with session.begin():
-                res = (
-                    session.query(
-                        self.table_class,
-                        self.table_class.embedding.l2_distance(embedding),
-                        # type: ignore
-                    )
-                    .order_by(self.table_class.embedding.l2_distance(embedding))
-                    .limit(limit)
-                )  # type: ignore
+                query = session.query(
+                    self.table_class,
+                    self.table_class.embedding.l2_distance(embedding),
+                ).order_by(self.table_class.embedding.l2_distance(embedding))
```

Comment:
I'm not a SQL expert, but I'm assuming it's ok that order_by comes before the filter?

Reply:
Yeah the generated SQL statement still looked correct to me when inspecting it